app/waf/src/models/profiles/http_compress.model.ts

```diff
@@ -0,0 +1,376 @@
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 4096,
+      openapi: {
+        type: 'integer',
+        minimum: 256,
+        maximum: 32768,
+      },
+    },
+    as3: {},
+  })
+  bufferSize?: number;
+
+  @property({
+    type: 'array',
+    itemType: 'string',
+    required: false,
+    default: [],
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: ['application/json', 'text/xml'],
+      openapi: {
+        items: {
+          type: 'string',
+        },
+      },
+    },
+    as3: {},
+  })
+  contentTypeExcludes?: string;
+
+  @property({
+    type: 'array',
+    itemType: 'string',
+    required: false,
+    default: ['text/', 'application/(xml|x-javascript)'],
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: ['application/json', 'text/xml'],
+      openapi: {
+        items: {
+          type: 'string',
+        },
+      },
+    },
+    as3: {},
+  })
+  contentTypeIncludes?: string;
+
+  @property({
+    type: 'boolean',
+    required: false,
+    default: true,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: true,
+      openapi: {
+        enum: [true, false],
+      },
+    },
+    as3: {},
+  })
+  cpuSaver?: boolean;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 90,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 90,
+      openapi: {
+        type: 'integer',
+        minimum: 15,
+        maximum: 99,
+      },
+    },
+    as3: {},
+  })
+  cpuSaverHigh?: number;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 75,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 75,
+      openapi: {
+        type: 'integer',
+        minimum: 10,
+        maximum: 95,
+      },
+    },
+    as3: {},
+  })
+  cpuSaverLow?: number;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 1,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 1,
+      openapi: {
+        type: 'integer',
+        minimum: 1,
+        maximum: 9,
+      },
+    },
+    as3: {},
+  })
+  gzipLevel?: number;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 8,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 8,
+      openapi: {
+        type: 'integer',
+        minimum: 1,
+        maximum: 256,
+      },
+    },
+    as3: {},
+  })
+  gzipMemory?: number;
+
+  @property({
+    type: 'number',
+    required: false,
+    default: 16,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: 16,
+      openapi: {
+        type: 'integer',
+        minimum: 1,
+        maximum: 128,
+      },
+    },
+    as3: {},
+  })
+  gzipWindowSize?: number;
```

Comment:
It should always have a value, so it does not need this '?'.

Reply:
removed